server.py

```diff
@@ -182,4 +182,9 @@ def file_upload(code, number):
         return "Solved! Great Job! "
 
 
+@app.error(404)
+def error404(error):
+    return template("error404.html")
```

Comment:
maybe also show information about why it's not found? For example 404 happens when the contest is not found

Reply:
i i have showed a default msg for error 404 and a link to go back to home directory

Reply:
if you don't want it like this then tell me what should i do to improve it??

Reply:
well, if you're handling the 404 error I would want to know why it's 404. For example is the route not available/is the contest code wrong/is the question not a part of the contest etc.
The template might have space for reasons specified by the error.

Reply:
yes but from where i will get this reason??

Reply:
from where i will get this reason??

Reply:
you can explore the `error` object that you are taking as argument. Probably somewhere in that? The advantage with bottle is that you can go read it's code since it's a single file in case you are not able to find something in the docs. For example, the error object you are getting in the function is the [HTTPError Object](https://github.com/bottlepy/bottle/blob/master/bottle.py#L1961)

Reply:
thank you i will sort out that but i am not able to include the css with rebase? can you help me in that

Reply:
sure. Why don't you open parallel merge requests with all the variations you have tried for inserting css with a rebase

Reply:
we can discuss your approaches in those merge reqests

Reply:
ok